### PR TITLE
[feat] selectively add SF / PH terms in Kanamori

### PR DESCRIPTION
**Pull request by the-hampel**
This PR allows 
* to select specifically spin-flip and / or pair-hopping terms are added in the construction of the `h_int_kanamori` many-body operator. Additionally I run auto format on the python file. 
* fixes `x_data_view`: for MeshReFreq objects it would only give back a np array of mesh points for `X`
* extend `x_data_view` to MeshImTime
* speed up `x_data_view` by removing for loop to extract `X` values

### CODE REVIEW

**Comment by Wentzell** [comment]
Thank you @the-hampel for the changes.

Can you please separate the code changes from the formatting changes in the commit history?

**Comment by the-hampel** [comment]
The commits are now separate into three parts: 1) formatting, 2) feature in hamiltonians.py, and 3) fix for `x_data_view` .

File: python/triqs/gf/gf.py [hunk 1]
```diff
@@ -806,7 +806,14 @@ def x_data_view(self, x_window=None, flatten_y=False):
             If flatten_y is True and dim is (1, 1, *) it returns a 1d numpy array.
         """
 
-        X = [x.imag for x in self.mesh] if isinstance(self.mesh, meshes.MeshImFreq) else [x for x in self.mesh]
+        if isinstance(self.mesh, meshes.MeshReFreq):
+            X = np.linspace(self.mesh.w_min, self.mesh.w_max, len(self.mesh))
```
Comment (Wentzell) [inline python/triqs/gf/gf.py]: I don't think it is guaranteed that this generates the same floating point numbers as `[x.real for x in self.mesh]` ?

File: python/triqs/gf/gf.py [hunk 2]
```diff
@@ -806,7 +806,14 @@ def x_data_view(self, x_window=None, flatten_y=False):
...
+        elif isinstance(self.mesh, meshes.MeshImFreq):
+            X = np.linspace(self.mesh(self.mesh.first_index()).imag, self.mesh(self.mesh.last_index()).imag, len(self.mesh))
```
Comment (Wentzell) [inline python/triqs/gf/gf.py]: Same as above

File: python/triqs/gf/gf.py [hunk 3]
```diff
@@ -806,7 +806,14 @@ def x_data_view(self, x_window=None, flatten_y=False):
...
+        elif isinstance(self.mesh, meshes.MeshImTime):
+            X = np.linspace(0, self.mesh.beta, len(self.mesh))
```
Comment (Wentzell) [inline python/triqs/gf/gf.py]: Same as above

**Review by Wentzell — CHANGES_REQUESTED** [review]
Dear @the-hampel

Thank you for these improvements! See comments below.

File: python/triqs/gf/gf.py (hunk 1 above)
Comment (the-hampel) [reply python/triqs/gf/gf.py]: hm, yes that is probably true. Are you concerned about exact equality of the values? Is it necessary? But okay then we need to loop again right? I hoped to make this method also a bit faster by avoiding for loops.

Comment (Wentzell) [reply python/triqs/gf/gf.py]: I have moved the generation of the values array to C++, please have a look at the changes.

**Review by the-hampel — COMMENTED** [review]
Thank you @Wentzell This looks good to me.

**Pull request merged**